## Практическое задание. Тестирование гипотез 

### 1. Когда используется критерий Стьюдента, а когда Z –критерий?

Z критерий:
1. Нормальность.
2. Сигма генеральной совокупности известна. 

t (критерий Стьюдента)
1. Нормальность.
2. Равенство дисперсий.
3. Сигма генеральной совокупности неизвестна.

### 2. Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм. 

In [2]:
import numpy as np 
import pylab 
import scipy.stats as stats

H0: mu = mu0

H1: mu > mu0 

In [28]:
mu = 17.5
mu0 = 17
sig = 2 #square root of 4 mmˆ2 
n = 100

zn = (mu - mu0)/(sig / np.sqrt(n))
zn

2.5

In [29]:
print(f'Z-value equals to: {round(zn, 1)}')

Z-value equals to: 2.5


In [30]:
a = 0.05

zt = stats.norm.ppf(1 - a)
zt

1.6448536269514722

In [31]:
print(f'Table z value: {round(zt, 4)}')

Table z value: 1.6449


In [33]:
zn < zt # zn(2.5) is greater than zt(1.6449) 

False

#### H0 отвергается так как zn < zt. 

### 3. Проведите тест гипотезы. Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%? (Провести двусторонний тест.)

H0: mu = mu0

H1: mu ≠ mu0 

In [34]:
pachk = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
pachk

array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

In [35]:
np.mean(pachk)
print(f'Mean weight value of packages: {round(np.mean(pachk))}')

Mean weight value of packages: 198


In [36]:
np.std(pachk, ddof = 1)

4.453463071962462

In [27]:
len(pachk)

10

In [ ]:
# t = (np.mean(pachk)  - exp.value) / (np.std / np.sqrt(n))
# t

In [40]:
t = (np.mean(pachk) - 200) / (np.std(pachk, ddof = 1) / np.sqrt(10))
print(f'Observed t-value: {round(t, 4)}')

Observed t-value: -1.0651


In [ ]:
#stats.ttest_lsamp(x, 10) - One sample t-test
#Ttest_lsampleResult(statistics = , pvalue = )

In [45]:
alpha = 0.005

tabt = stats.t.ppf(1 - alpha, n - 1)
print(f'Table t-value: {round(tabt, 4)}')

Table t-value: 2.6264


In [44]:
np.abs(t) < tabt

True

Так как t(вычисленное значение t-критерия) < tabt(табличное значение t-критерия), то гипотеза H1 отвергается, а гипотеза H0 принемается.

### 4. Pешать с помощью функции. Есть ли статистически значимые различия в росте дочерей?

Рост матерей 172, 177, 158, 170, 178,175, 164, 160, 169

Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160

In [12]:
x = np.array([172, 177, 158, 170, 178, 175, 164, 160, 169])
y = np.array([173, 175, 162, 174, 175, 168, 155, 170, 160])
# the arrays should contain the same number of values 

In [13]:
stats.ttest_rel(x, y) # function for dependent samples t-test

TtestResult(statistic=0.559522990335608, pvalue=0.5911212354055175, df=8)

Значение p-value = 0.591 больше значения α = 0.05. 
Так как 0.591 > 0.05, гипотеза H1 отвергается, а гипотеза H0 принемается. Значит, статистической разницы в среднем ростом взрослых дочерей и матерей нет. 